src/filter-component.js

```diff
@@ -108,8 +108,9 @@ FilterComponent.prototype._checkFields =
         }
 
         const allowed_values = self[name];
-        if (allowed_values) {
-            if (!allowed_values.map(match_func)) {
+        if (allowed_values && allowed_values.length > 0) {
+            const anyMatch = allowed_values.map(match_func).reduce((a, b) => a || b);
```

Comment:
```suggestion
            const anyMatch = allowed_values.some(match_func);
```

would be simpler unless I am missing something

Reply:
No, you are right. I didn't know something like `some` existed.